pykokkos/lib/ufuncs.py

```diff
@@ -1381,6 +1389,270 @@ def exp2(view):
     return out
 
 
+@pk.workunit
+def sum_impl_1d_double(tid: int, acc: pk.Acc[pk.double], viewA: pk.View1D[pk.double]):
+    acc += viewA[tid]
+
+
+def sum(viewA, axis=None):
+    if(axis is not None):
+        res = np.sum(viewA, axis=axis)
+        view = pk.View(res.shape, pk.double)
+        view[:] = res
+
+        return view
+
+    if str(viewA.dtype) == "DataType.double":
+        return pk.parallel_reduce(
+            viewA.shape[0],
+            sum_impl_1d_double,
+            viewA=viewA)
+
+
+def argmax(view, axis):
+    res = np.argmax(view, axis=axis)
+    view = pk.View(res.shape, pk.int32)
+    view[:] = res
+
+    return view
+
+def unique(view):
+    res = np.unique(view)
+    view = pk.View(res.shape, pk.double)
+    view[:] = res
+
+    return view
+
+def var(view, axis):
+    res = np.var(view, axis=axis)
+    view = pk.View(res.shape, pk.double)
+    view[:] = res
+
+    return view
+
+# @pk.workunit
+# def mean0___impl_1d_double(tid: int, viewA: pk.View2D[pk.double], out: pk.View1D[pk.double]):
+#     out[tid] = 0
+#     for i in range(viewA.extent(0)):
+#         out[tid] = out[tid] + (viewA[tid][i] / viewA.extent(0))
+
+
+# @pk.workunit
+# def mean1_____impl_1d_double(tid: int, viewA: pk.View2D[pk.double], out: pk.View1D[pk.double]):
+#     x = 0
+#     for i in range(viewA.extent(1)):
+#         x += (viewA[i][tid] / viewA.extent(1))
+#     out[tid] = x
+
+
+
+def mean(view, axis):
+    viewA = pk.View(view.shape, pk.double)
+    viewA[:] = view
+
+    if str(viewA.dtype) == "DataType.double":
+        if (axis == 0):
+            sum_view = sum(viewA, axis=0)
+            return divide_num(sum_view, viewA.shape[0]) 
+        else:
+            sum_view = sum(viewA, axis=1)
+            return divide_num(sum_view, viewA.shape[1])
+    else:
+        raise RuntimeError("Incompatible Types")
```

Comment:
@ShahzadUmair I think you'll fix this as discussed?

Reply:
Done